### ISSUE TITLE
Compile PyTorch with OpenBLAS/LAPACK

### ISSUE THREAD

**Issue by mahmoodn** [pytorch/pytorch#23650]
I want to install PyTorch with an installed OpenBLAS. However, I can not find any guide on how to tell `python setup.py install` to use /path/to/libopenblas.a. Any idea?

**Comment by vishwakftw** [comment]
PyTorch detects the available BLAS library using CMake.

Please refer https://github.com/pytorch/pytorch/issues/8561 for how this is done.

**Comment by mahmoodn** [comment]
Since I am using `python setup.py install`, I am not sure what should I do with cmake.
Do you mean, if I run `sudo apt install liblapack`, that `setup.py` will automatically detects that?

**Comment by vishwakftw** [comment]
Sorry, let me rephrase. CMake automatically finds out available BLAS/LAPACK libraries, so ideally you should be able to just run `python setup.py install`.

cc: @soumith (just to ensure I didn’t misquote any details)

**Comment by vishwakftw** [comment]
If you have installed OpenBLAS at a non-standard location, please export the location to `OpenBLAS_HOME` as:
```
export OpenBLAS_HOME=path/to/OpenBLAS
```
and then run python setup.py install

**Comment by mahmoodn** [comment]
Thank you. I installed `apt install liblapack-dev` and ran `setup.py install` again and error is now fixed.

**Comment by amithm3** [comment]
@vishwakftw @mahmoodn can i just do OpenBLAS_HOME=path/to/OpenBLAS and pip install torch, for it to work or is there something else I need to do?
